src/mono/mono/component/debugger-agent.c

```diff
@@ -7940,6 +7948,17 @@ domain_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 		buffer_add_objid (buf, o);
 		break;
 	}
+	case MDBGPROT_CMD_APPDOMAIN_GET_TYPE: {
+		MonoClass *klass;
+		domain = decode_domainid (p, &p, end, NULL, &err);
+		MonoTypeEnum type = decode_int (p, &p, end);
+		klass = decode_typeid (p, &p, end, NULL, &err);
+		int rank = decode_int (p, &p, end);
+		if (type == MONO_TYPE_SZARRAY || type == MONO_TYPE_ARRAY)
+			klass = mono_class_create_array (klass, rank);
+		buffer_add_typeid (buf, domain, klass);
+		break;
```

Comment:
This is a weird command.  It always sends a "type" enum, a base type and a rank, but if "type" is not array it just sends base the basetype (and ignores the rank) and otherwise it creates an array?  Does it really have to behave like this?

(Also will you need to support array bounds later?)